<?xml version="1.0"?>

<launch>
  <arg name="camera_name" default="zed2_up"/>
  <arg name="camera_frame" default="zed2_up_left_camera_optical_frame" />
  <arg name="image_topic" default="/zed2_up/zed_up_node/rgb/image_rect_color" />
  <arg name="info_topic" default="/zed2_up/zed_up_node/rgb/camera_info" />

  <arg name="image_topic_synced" default="/zed2_up/zed_up_node/synced_rgb/image_rect" />
  <arg name="info_topic_synced" default="/zed2_up/zed_up_node/synced_rgb/camera_info" />

  <arg name="tag_file" default="tags.yaml" />

  <!-- Set parameters -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros" clear_params="true" output="screen" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg image_topic_synced)" />
    <remap from="camera_info" to="$(arg info_topic_synced)" />
    <rosparam command="load" file="$(find arm_calibration)/config/settings.yaml" />
    <rosparam command="load" file="$(find arm_calibration)/config/$(arg tag_file)" />
    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />
  </node>

  <!-- Launch the calibration node -->
  <!-- pkg: arm_calibration sync_images.py -->
  <node pkg="arm_calibration" type="sync_images.py" name="sync_images" output="screen" >
    <remap from="synced_camera_info" to="$(arg info_topic_synced)" />
    <remap from="camera_info" to="$(arg info_topic)" />
    <remap from="synced_camera_image" to="$(arg image_topic_synced)" />
    <remap from="camera_image" to="$(arg image_topic)" />
  </node>
</launch>
